services/worker-manager/src/providers/azure.js

```diff
@@ -165,6 +165,17 @@ class AzureProvider extends Provider {
         workerConfig: cfg.workerConfig || {},
       })).toString('base64');
 
+      // Disallow users from naming diss
+      // required
+      let osDisk = {..._.omit(cfg.storageProfile.osDisk, ['name'])};
```

Comment:
We should probably throw an error in this case. Let's leave that to a followup (I think there are ways to disallow a property in json-schema, so that's probably the better fix).